.github/workflows/test.yaml

```
@@ -484,14 +484,13 @@ jobs:
       run: |
         tox -e narrativedocs
 
-
-  # Used for various release automation.
+  # Used for various release automation, but not publishing
```

Maybe explain why we have the separate job for publishing.

```suggestion
  # Used for various release automation, but not for publishing, as publishing needs extra permissions for the API token.
  # So we isolate the actual publish step in a separate job.
```